p2p-tests/contributor-fork-replication-1-test.ts

```diff
@@ -0,0 +1,177 @@
+    await maintainer.proxyClient.project.create({
+      repo: {
+        type: "new",
+        path: path.join(dataPath, "maintainer-checkouts"),
+        name: "my-fancy-project",
+      },
+      description: "",
+      defaultBranch: "main",
+    });
+  });
+
+  await maintainer.stop();
+  seed.start();
+  maintainer.start();
+
+  await withRetry(async () => {
+    const result = radCli({
+      radHome: seed.radHome,
+      args: ["identities", "project", "get", "--urn", projectUrn],
+    });
+
+    strictAssert.deepStrictEqual(result, {
+      urn: projectUrn,
+      payload: {
+        "https://radicle.xyz/link/identities/project/v1": {
+          name: "my-fancy-project",
+          description: "",
+          default_branch: "main",
+        },
+      },
+    });
+  });
+
+  await maintainer.stop();
+  contributor.start();
+
+  await withRetry(async () => {
+    await contributor.proxyClient.project.requestSubmit(projectUrn);
+  });
+
+  await withRetry(async () => {
+    const project = await contributor.proxyClient.project.get(projectUrn);
+
+    strictAssert.deepStrictEqual(project, {
+      urn: projectUrn,
+      metadata: {
+        name: "my-fancy-project",
+        description: "",
+        defaultBranch: "main",
+        maintainers: [maintainerUrn],
+      },
+      stats: { commits: 1, branches: 0, contributors: 1 },
+    });
+  });
+
+  const contributorForkPath = await contributor.proxyClient.project.checkout(
+    projectUrn,
+    {
+      path: path.join(dataPath, "contributor-checkouts"),
+      peerId: maintainerPeerId,
+    }
+  );
+
+  commit({ author: contributor.namespace, checkoutPath: contributorForkPath });
+  pushRad({
+    radHome: contributor.radHome,
+    checkoutPath: contributorForkPath,
+    keyPassphrase: contributor.passphrase,
+  });
+
+  await withRetry(async () => {
+    const result = radCli({
+      radHome: seed.radHome,
+      args: ["identities", "project", "tracked", "--urn", projectUrn],
+    });
+
+    strictAssert.strictEqual(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      result.find((x: any) => x.peerId === contributorPeerId).status.type,
+      "replicated"
+    );
+  });
+  await contributor.stop();
+
+  maintainer.start();
+
+  await withRetry(async () => {
+    await maintainer.proxyClient.project.peerTrack(
+      projectUrn,
+      contributorPeerId
+    );
+  });
+
+  await withRetry(async () => {
+    const commitList = await maintainer.proxyClient.source.commitsGet({
+      projectUrn,
+      peerId: contributorPeerId,
+      revision: { type: source.RevisionType.Branch, name: "main" },
+    });
+
+    strictAssert.deepStrictEqual(commitList?.stats, {
+      commits: 2,
+      branches: 1,
+      contributors: 2,
+    });
+  });
+}
+
+runTestcase({
+  testcase,
+  networkScript: "star-topology.sh",
+  dataDirName: "contributor-fork-replication-1-test",
+  teardown: async (): Promise<void> => {
+    await maintainer.kill();
```

Comment:
I’m not sure it is necessary to manually kill the processes spawned by `execa`. The library [is set up](https://www.npmjs.com/package/execa#cleanup) to kill all children when the node process exits. 

In any case, I found the previous solution with a global process register to be easier to use: There was only one function call to clean up all processes and we did not rely on this functionality cluttering the classes.

Reply:
Thanks for pointing this out. I was looking for how to simplify this. What I was missing was a `process.exit(0)` at the end of the test case, that took care of cleaning up all the processes.

066fdc217025ce475ddd1eeab508e3d8e83d121b